plugins/x86/x86_prefix.ml

```diff
@@ -0,0 +1,53 @@
+open Core_kernel.Std
+open Or_error
+open Bap.Std
+
+type t = [
+  | `_F0 (** LOCK                *)
+  | `_F2 (** REPNE               *)
+  | `_F3 (** REP/REPE            *)
+  | `_2E (** CS segment override *)
+  | `_36 (** SS segment override *)
+  | `_3E (** DS segment override *)
+  | `_26 (** ES segment override *)
+  | `_64 (** FS segment override *)
+  | `_65 (** GS segment override *)
+  | `_66 (** operand override    *)
+  | `_67 (** address override    *)
+] [@@deriving bin_io, sexp, compare, enumerate]
+
+let to_known_prefix = function
+  | 0xF0 -> Some `_F0
+  | 0xF2 -> Some `_F2 
+  | 0xF3 -> Some `_F3
+  | 0x2E -> Some `_2E
+  | 0x36 -> Some `_36
+  | 0x3E -> Some `_3E
+  | 0x26 -> Some `_26
+  | 0x64 -> Some `_64
+  | 0x65 -> Some `_65
+  | 0x66 -> Some `_66
+  | 0x67 -> Some `_67
+  | _ -> None
+
+let deref: Memory.t -> word Seq.t = fun mem ->
+  Seq.unfold_step ~init:(Memory.min_addr mem)
+    ~f:(fun addr ->
+        match Memory.get ~addr mem with
+        | Ok word -> Seq.Step.Yield (word, Word.succ addr)
+        | Error _ -> Seq.Step.Done)
+
+let get mem =
+  let rec loop bytes prefs =
```

Comment:
It can be expressed more clearly using `Seq.fold`, or even better with few maps and `take_until`, e.g.,
```
Seq.(bytes >>| Word.to_int >>| ok_exn >>| to_known_prefix >>| take_until ~f:not_none) 
```